dll/ntdll/csr/connect.c

```diff
@@ -366,11 +371,18 @@ CsrClientCallServer(IN OUT PCSR_API_MESSAGE ApiMessage,
     ULONG PointerCount;
     PULONG_PTR OffsetPointer;
 
+    /* Make sure the length is valid */
+    if (DataLength > (MAXSHORT - sizeof(CSR_API_MESSAGE)))
```

Comment:
s/MAXSHORT/MAXUSHORT/

Reply:
No, because ```ApiMessage->Header.u1.s1.TotalLength``` is a ```CSHORT```, which is signed.

Reply:
I wonder why they used CSHORTs.... a negative data length just doesn't make sense at all.